cpu/cm4-tool/tool.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT117x M4 cpu core tool
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <string.h>
+#include <sys/msg.h>
+#include <imxrt117x-cm4.h>
+
+
+static const char blinky[] = {
+#include "blinky.h"
+};
+
+
+int main(int argc, char *argv[])
+{
+	msg_t msg;
+	imxrt117xM4DevCtli_t *i = (void *)msg.i.raw;
+	imxrt117xM4DevCtlo_t *o = (void *)msg.o.raw;
+	int opt, usage = 0, file = 0, example = 0;
+	char *path = NULL;
+	oid_t driver;
+
+	if (argc == 1)
+		return 0;
+
+	if (lookup("/dev/cpuM40", NULL, &driver) < 0) {
+		fprintf(stderr, "imxrt117x-cm4 driver not found!\n");
+
+		return -1;
+	}
+
+	while ((opt = getopt(argc, argv, "f:e")) >= 0) {
+		switch (opt) {
+		case 'f':
+			file = 1;
+			if ((path = strdup(optarg)) == NULL) {
+				fprintf(stderr, "Out of memory\n");
+				return -1;
+			}
+			break;
+
+		case 'e':
+			example = 1;
+			break;
+
+		default:
+			usage = 1;
+			break;
+		}
+	}
+
+	if (usage || (file && example) || (!file && !example)) {
```

Comment:
```suggestion
	if (usage || !(file ^ example)) {
```

Reply:
I've thought about it, but the is only bitwise version of EOR in C, I don't feel comfortable with using bitwise logic on bool variables. Code as is will work with e.g. `file = 2` and `example = 1`, but `^` wouldn't.

Reply:
A logical `xor` can be done like this: `(!file == !example)`

Reply:
I think that the original verion is more readable, but thank you for this method